src/Microsoft.ML.TimeSeries/FftUtils.cs

```diff
@@ -166,35 +166,35 @@ private enum ConfigValue
         private const string DllProxyName = "MklProxyNative";
 
         // See: https://software.intel.com/en-us/node/521976#8CD904AB-244B-42E4-820A-CC2376E776B8
-        [DllImport(DllProxyName, EntryPoint = "MKLDftiCreateDescriptor", CallingConvention = CallingConvention.Cdecl, CharSet = CharSet.Auto)]
+        [DllImport(DllProxyName, EntryPoint = "MKLDftiCreateDescriptor", CharSet = CharSet.Auto)]
         private static extern int CreateDescriptor(out IntPtr desc, ConfigValue precision, ConfigValue domain, int dimension, int length);
 
         // See: https://software.intel.com/en-us/node/521977
-        [DllImport(DllName, EntryPoint = "DftiCommitDescriptor")]
+        [DllImport(DllName, CallingConvention = CallingConvention.Cdecl, EntryPoint = "DftiCommitDescriptor")]
         private static extern int CommitDescriptor(IntPtr desc);
 
         // See: https://software.intel.com/en-us/node/521978
-        [DllImport(DllName, EntryPoint = "DftiFreeDescriptor")]
+        [DllImport(DllName, CallingConvention = CallingConvention.Cdecl, EntryPoint = "DftiFreeDescriptor")]
         private static extern int FreeDescriptor(ref IntPtr desc);
 
         // See: https://software.intel.com/en-us/node/521981
-        [DllImport(DllProxyName, EntryPoint = "MKLDftiSetValue", CallingConvention = CallingConvention.Cdecl, CharSet = CharSet.Auto)]
+        [DllImport(DllProxyName, EntryPoint = "MKLDftiSetValue", CharSet = CharSet.Auto)]
```

Comment:
It looks we are using MKL. Is it not a `Cdecl`?

Reply:
this is a proxy which we build ourselves, so it would use default convention. We can't change Mkl (which is not proxy) so for pure Mkl we have to use cdecl.

---
In reply to: [239658702](https://github.com/dotnet/machinelearning/pull/1833#discussion_r239658702) [](ancestors = 239658702)